packages/sdk/src/contracts/bridge.contract.ts

```diff
@@ -0,0 +1,427 @@
+   * @param chainIds The list of network chainIds to get the results from, if nothing is provided the default are the networks defined in the config.
+   * @returns The HolographOperator contract address per network.
+   */
+  async getOperatorByNetworks(chainIds?: number[]): Promise<HolographByNetworksResponse> {
+    const results: HolographByNetworksResponse = {}
+    let networks = getSelectedNetworks(this.networks, chainIds)
+
+    for (const network of networks) {
+      results[network.chain] = await this._getContractFunction(network.chain, 'getOperator')
+    }
+
+    return results
+  }
+
+  /**
+   * @readonly
+   * Gets the Holograph Registry module.
+   * This module stores a reference for all deployed holographable smart contracts.
+   * @param chainId The chain id of the network to get the result from.
+   * @returns The Holograph Registry contract address in the provided network.
+   */
+  async getRegistry(chainId: number) {
+    return this._getContractFunction(chainId, 'getRegistry')
+  }
+
+  /**
+   * @readonly
+   * Gets the Holograph Registry module.
+   * This module stores a reference for all deployed holographable smart contracts.
+   * @param chainIds The list of network chainIds to get the results from, if nothing is provided the default are the networks defined in the config.
+   * @returns The Holograph Registry contract address per network.
+   */
+  async getRegistryByNetworks(chainIds?: number[]): Promise<HolographByNetworksResponse> {
+    const results: HolographByNetworksResponse = {}
+    let networks = getSelectedNetworks(this.networks, chainIds)
+
+    for (const network of networks) {
+      results[network.chain] = await this._getContractFunction(network.chain, 'getRegistry')
+    }
+
+    return results
+  }
+
+  /**
+   * @readonly
+   * Gets the fees associated with sending specific payload.
+   * It will provide exact costs on protocol and message side, combine the two to get total.
+   * @param chainId The chain id of the network to send the transaction.
+   * @param toChain The Holograph chain id of destination chain for payload.
+   * @param gasLimit The amount of gas to provide for executing payload on destination chain.
+   * @param gasPrice The maximum amount to pay for gas price, can be set to 0 and will be chose automatically.
+   * @param crossChainPayload The entire packet being sent cross-chain.
+   * @returns stringifiedFeesArray The compound fees [hlfFee, msgFee, dstGasPrice]:
+   * hlgFee: The amount (in wei) of native gas token that will cost for finalizing job on destination chain.
+   * msgFee: The amount (in wei) of native gas token that will cost for sending message to destination chain.
+   * dstGasPrice: The amount (in wei) that destination message maximum gas price will be.
+   */
+  async getMessageFee(
+    chainId: number,
+    toChain: number,
+    gasLimit: bigint,
+    gasPrice: bigint,
+    crossChainPayload: string | Buffer,
```

Comment:
Instead of `string` you can use `Hex` because a type `bytes` is a `0x${string}`. 
To help define the types you can use this:

`type _PrimitiveType = AbiParameterToPrimitiveType<{name: 'crossChainPayload'; type: 'bytes'}> `

The inferred type here is probably the best choice.

Reply:
Yesss, I've been using this a lot